File: src/app/enums/investment-status.ts
// src/app/enums/investment-status.ts

export enum InvestmentStatus {
  OPEN = 'OPEN',
  CLOSED = 'CLOSED',
}

// For *ngFor and filtering
export const INVESTMENT_STATUS_VALUES: InvestmentStatus[] =
  Object.values(InvestmentStatus);

File: Backend/src/main/java/com/alpha/alphavault/enums/InvestmentStatus.java
/**
 * ================================================================
 *  Coded by Mohamed Dhaoui for Alpha Vault - Financial System
 *  Enum: InvestmentStatus (API-safe as STRING)
 * ================================================================
 */
package com.alpha.alphavault.enums;

import com.fasterxml.jackson.annotation.JsonFormat;

@JsonFormat(shape = JsonFormat.Shape.STRING)
public enum InvestmentStatus {
    OPEN,
    CLOSED
}
